web_edit_user_filter/static/src/js/searchBar.esm.js

```diff
@@ -0,0 +1,159 @@
+/** @odoo-module **/
+
+import {SearchBar} from "@web/search/search_bar/search_bar";
+import {patch} from "@web/core/utils/patch";
+import {qweb} from "web.core";
+import {_t} from "@web/core/l10n/translation";
+import {useService} from "@web/core/utils/hooks";
+
+patch(SearchBar.prototype, "web_edit_user_filter.SearchBar", {
+    setup() {
+        this.notificationService = useService("notification");
+        var self = this;
+        this._super(...arguments);
+        $("body").on("click", ".o_searchview_facet", function (ev) {
+            $(ev.currentTarget).popover("dispose");
+            if ($(ev.currentTarget).hasClass("o_facet_remove")) {
+                return;
+            }
+            var facet_type = $(this).attr("data-type");
+            var facetId = $(this).attr("data-gp");
+            self._process_filters($(this), facet_type, facetId);
+        });
+    },
+
+    _process_filters($el, facet_type, facetId) {
+        var searchItems = this.env.searchModel.getSearchItems(
+            (facet) => facet.type === facet_type
+        );
+
+        var selectedFacet = searchItems.filter(
+            (f) => f.groupId == facetId && f.isActive === true
+        );
+
+        if (!selectedFacet.length) {
+            return;
+        }
+        if (facet_type === "favorite") {
+            var FavFacets = [];
+            var currentFacet = this.env.searchModel.getSearchItems(
+                (f) => f.type === facet_type && f.groupId == facetId
+            );
```

Comment:
**comment (non-blocking)**: I think in this case using `searchItems.filter` was better, because you're filtering a subset of the `searchItems` you had already filtered beforehand, instead of searching from every filter in the model.

On the other hand, if `searchItems` is already filtered by `facet_type` and you use `.filter` here, you can drop the `f.type === facet_type` condition because it was already checked earlier.

Reply:
Updated as suggested